crmsh/completers.py

```diff
@@ -49,22 +49,19 @@ def resources(args=None):
     if cib_el is None:
         return []
     nodes = xmlutil.get_interesting_nodes(cib_el, [])
-    res = [x.get("id") for x in nodes if xmlutil.is_resource(x)]
-    if args and args[0] in ['promote', 'demote']:
-        return list(filter(xmlutil.RscState().is_ms, res))
+    rsc_id_list = [x.get("id") for x in nodes if xmlutil.is_resource(x)]
     if args and args[0] == "started":
-        return list(filter(xmlutil.RscState().is_running, res))
+        return [item for item in rsc_id_list if xmlutil.RscState().is_running(item)]
     if args and args[0] == "stopped":
-        for item in filter(xmlutil.RscState().is_running, res):
-            res.remove(item)
-    return res
+        return [item for item in rsc_id_list if not xmlutil.RscState().is_running(item)]
+    return rsc_id_list
 
 
-def resources_started(args):
+def resources_started(args=None):
```

Comment:
Since all the caller are calling without `arg` and always call `resources(["started"])`, so maybe no need to have `args=None`, use `resources_started()` should enough.

Reply:
But the caller `@command.completers(compl.resources_started)`, inside it:
```
def completers(*fns):
    '''
    Use to set a list of positional tab completers for the command.
    Each completer gets as its argument the command line entered so far,
    and returns a list of possible completions.
    '''
    def cfn(args):
        nargs = len(args) - 1
        if nargs == 0:
            return [args[0]]
        if nargs <= len(fns):
            return fns[nargs-1](args)
        return []

    def inner(fn):
        setattr(fn, '_completer', cfn)
        return fn
    return inner
```

`fns[nargs-1](args)` it contains 1 positional argument